src/libcore/ops/range.rs

```diff
@@ -861,7 +861,7 @@ impl<'a, T: ?Sized + 'a> RangeBounds<T> for (Bound<&'a T>, Bound<&'a T>) {
 }
 
 #[stable(feature = "collections_range", since = "1.28.0")]
-impl<T> RangeBounds<T> for RangeFrom<&T> {
+impl<T: ?Sized> RangeBounds<T> for RangeFrom<&T> {
```

Comment:
Isn't it *technically* a breaking change to expand the scope of a blanket impl? e.g. if someone had already implemented this for their own unsized `Foo`, that would now conflict.

Reply:
Yes. We usually categorize it as a so-called “minor” breaking change: https://rust-lang.github.io/rfcs/1105-api-evolution.html#minor-change-implementing-any-non-fundamental-trait

Reply:
That's minor regarding dispatch ambiguity, but blanket impls cause actual conflicts.

https://github.com/rust-lang/rfcs/blob/master/text/1023-rebalancing-coherence.md
https://rust-lang.github.io/rfcs/2451-re-rebalancing-coherence.html

> As such, RFC #1105 is amended to remove the statement that implementing a non-fundamental trait is a minor breaking change, and states that adding any blanket impl for an existing trait is a major breaking change, using the definition of blanket impl given above.

That doesn't address *expanding* a blanket impl by relaxing constraints, as here, but I think the problem is the same and should be considered a major breaking change. It's particularly relevant that `Sized` **is** a fundamental trait, so coherence allows child crates to assume that their unsized type will never overlap with this current `Sized` impl.

The reason I said "*technically*" before was that I doubt anyone would actually have a conflicting impl in this specific case, but it's possible.

Reply:
(Getting more and more off-topic, but: I think we should not have RFCs that say “RFC X is amended to…” without actually applying the change to the other file. More generally, “facts” that are out of date should not be found on https://rust-lang.github.io/rfcs/.)